lib/configgin.rb

```diff
@@ -59,23 +60,43 @@ def generate_jobs(job_configs, templates)
 
   # Set the exported properties and their digests, and return the digests.
   def patch_job_metadata(jobs)
+    pod = kube_client.get_pod(@self_name, kube_namespace)
+
+    secret = Kubeclient::Resource.new
+    secret.metadata = {}
+    # Prefixing with pod.metadata.name is purely for human convenience/debugging.
+    secret.metadata.name = "#{pod.metadata.name}-#{pod.metadata.uid}"
+    secret.metadata.namespace = kube_namespace
+
+    # Make sure the secret gets removed when the pod is deleted.
+    secret.metadata.ownerReferences = [
+      {
+        :apiVersion => pod.apiVersion,
+        :blockOwnerDeletion => false,
+        :controller => false,
+        :kind => pod.kind,
+        :name => pod.metadata.name,
+        :uid => pod.metadata.uid,
```

Comment:
Out of curiosity, why are we using the rocket syntax? Can we not do `apiVersion: pod.apiVersion` etc? (I don't think we support Ruby old enough to not support it…)

Reply:
Idk, I don't have a preference either way. I can change it if you want me to.